File: backend/peliiculas/src/main/java/com/netflix/latam/peliiculas/Model/Compra.java
package com.netflix.latam.peliiculas.Model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "compras" )
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Compra {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "fecha_compra")
    private LocalDate fechaCompra;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "cliente_id")
    private Cliente cliente;

    @OneToMany(
            mappedBy = "compra",
            cascade = CascadeType.ALL,
            fetch = FetchType.LAZY
    )
    private List<DetalleCompra> detalles = new ArrayList<>();

    public void addDetalle(DetalleCompra detalle) {
        detalles.add(detalle);
        detalle.setCompra(this);
    };
}

File: frontend/src/Interfaces/Compra.ts
export interface DetalleCompra {
    id: number;
    peliculaNombre: string;
}

export interface Compra {
    id: number;
    clienteNombre: string;
    fechaCompra: string;
    detalles: DetalleCompra[];
}

export interface DetalleCompraRequest {
    peliculaId: number;
}

export interface CompraRequest {
    clienteId: number;
    fechaCompra: string;
    detalles: DetalleCompraRequest[];
}
